gm4_apple_trees/data/gm4_fruiting_trees/functions/tree/initialize.mcfunction

```diff
@@ -12,6 +12,6 @@ scoreboard players set $max_layer gm4_tree_layer -1
 function #gm4_fruiting_trees:tree/initialize
 
 # start generation, unless expansion force quit generation (e.g. due to missing dy).
-execute unless score $cancel_generation gm4_tree_layer matches 1.. run function gm4_fruiting_trees:tree/analyze_seed
+execute unless score $cancel_generation gm4_tree_layer matches 1.. at @s run function gm4_fruiting_trees:tree/analyze_seed
```

Comment:
Doesn't this already run at the sapling marker?

Reply:
Yes, but if you move the marker during the initialize function (line 12)—for example for a 2x2 tree—then the reference gets reset to the old location when actually spawning in the tree. This would ensure that the location is set correctly after the initialize functions are run.